app/src/main/java/org/fossasia/susi/ai/activities/MainActivity.java

```diff
@@ -269,62 +267,75 @@ public boolean checkMicInput() {
-            answer = getString(R.string.error_occurred_try_again);
-            isHavingLink = false;
-        }
 
-        //Check for map
-        try {
-            isMap = susiResponse.getAnswers().get(0).getActions().get(i).getType().equals("map");
-            datumList = susiResponse.getAnswers().get(0).getData();
-        } catch (Exception e) {
-            isMap = false;
-        }
+        actionType = susiResponse.getAnswers().get(0).getActions().get(i).getType();
+        datumList = null;
+        mapData = null;
+        webSearch = "";
+        isHavingLink = false;
 
-        //Check for piechart
-        try {
-            isPieChart = susiResponse.getAnswers().get(0).getActions().get(i).getType().equals("piechart");
-            datumList = susiResponse.getAnswers().get(0).getData();
-        } catch (Exception e) {
-            Log.d(TAG, e.getLocalizedMessage());
-            isPieChart = false;
-        }
+        switch(actionType) {
+            case Constant.ANCHOR :
+                try {
+                    boolean isAnchor = susiResponse.getAnswers().get(0).getActions().get(i).getType().equals("anchor");
+                    if(isAnchor)
+                        answer = "<a href=\"" +susiResponse.getAnswers().get(0).getActions().get(i).getAnchorLink() + "\">" + susiResponse.getAnswers().get(0).getActions().get(1).getAnchorText() + "</a>";
+                } catch (Exception e) {
+                    answer = getString(R.string.error_occurred_try_again);
+                }
+                break;
 
-        //Check for rss
-        try {
-            isSearchResult = susiResponse.getAnswers().get(0).getActions().get(i).getType().equals("rss");
-            datumList = susiResponse.getAnswers().get(0).getData();
-        } catch (Exception e) {
-            isSearchResult = false;
-        }
+            case Constant.ANSWER :
+                try {
+                    answer = susiResponse.getAnswers().get(0).getActions()
+                            .get(i).getExpression();
+                    List<String> urlList = extractUrls(answer);
+                    Log.d(TAG, urlList.toString());
+                    isHavingLink = urlList != null;
+                    if (urlList.size() == 0) isHavingLink = false;
+                } catch (Exception e ) {
+                    answer = getString(R.string.error_occurred_try_again);
+                    isHavingLink = false;
+                }
+                break;
 
-        //Check for websearch
-        try {
-            isWebSearch = susiResponse.getAnswers().get(0).getActions().get(i).getType().equals("websearch");
-            datumList = susiResponse.getAnswers().get(0).getData();
-            webSearch = susiResponse.getAnswers().get(0).getActions().get(1).getQuery();
-        } catch (Exception e) {
-            isWebSearch = false;
-            webSearch = "";
-        }
+            case Constant.MAP :
+                try {
```

Comment:
Here you used the same thing which I used to solve this problem but you rejected that and now you used same thing.You even not tell me about that.  I am working on this today. Please let me solve some issues. For your kind information I am also part of gsoc 2017 program. @mariobehling can you please see it.